### PR TITLE
process kernarg metadata using offset instead of align

### PR DESCRIPTION
**Pull request by david-salinas**
Change-Id: I12067177ffeb2d228c96f428405ca0ebe3e167ee

### CODE REVIEW

File: src/program_state.inl [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 #include "../include/hip/hcc_detail/code_object_bundle.hpp"
 #include "../include/hip/hcc_detail/hsa_helpers.hpp"
+#include "hip/hip_runtime.h"
```
Comment (emankov) [inline src/program_state.inl]: What part of the change needs this include?

File: include/hip/hcc_detail/functional_grid_launch.hpp [hunk 2]
```diff
@@ -87,14 +87,13 @@ inline hip_impl::kernarg make_kernarg(
                 "function");
     #endif
 
-    kernarg.resize(round_up_to_next_multiple_nonnegative(
-        kernarg.size(), size_align.alignment(n)) + size_align.size(n));
+        kernarg.resize(size_offset.offset(n) + size_offset.size(n));
```
Comment (aaronenyeshi) [inline include/hip/hcc_detail/functional_grid_launch.hpp]: Don't need the indent

File: src/program_state.inl (hunk 1 above)
Comment (aaronenyeshi) [reply src/program_state.inl]: I tried this patch, I am getting this error without this include:
```
In file included from /home/aaron/git/hip/src/program_state.cpp:3:
/home/aaron/git/hip/src/program_state.inl:653:37: error: no member named 'round_up_to_next_multiple_nonnegative' in namespace 'hip_impl'
            auto offset = hip_impl::round_up_to_next_multiple_nonnegative(current_offset,align);
```

File: include/hip/hcc_detail/functional_grid_launch.hpp (hunk 2 above)
Comment (david-salinas) [reply include/hip/hcc_detail/functional_grid_launch.hpp]: should be resolved in update

File: include/hip/hcc_detail/functional_grid_launch.hpp [hunk 3]
```diff
@@ -87,14 +87,13 @@ inline hip_impl::kernarg make_kernarg(
...
 
     std::memcpy(
-        kernarg.data() + kernarg.size() - size_align.size(n),
+        kernarg.data() + kernarg.size() - size_offset.size(n),
```
[the remark this answers is not in the document]
Comment (david-salinas) [reply include/hip/hcc_detail/functional_grid_launch.hpp]: fixed in the update

File: bin/hipcc [hunk 4]
```diff
@@ -342,7 +342,7 @@ my $runCmd = 1;
 my $buildDeps = 0;
 my $linkType = 1;
 my $setLinkType = 0;
-my $coFormatv3 = 0;
+my $coFormatv3 = 1;
```
[the remark this answers is not in the document]
Comment (david-salinas) [reply bin/hipcc]: removed in update

**Comment by david-salinas** [comment]
We're performing some additional/extensive testing.  We should not merge this until we get successful results back.

**Comment by mangupta** [comment]
Closing all existing pull requests that do not target the develop branch. In case this pull request is still valid, please rebase your changes against develop branch and resubmit.

**Pull request closed without merging**